plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -579,6 +605,7 @@ public LocatedTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
             return null;
         }
         verifySupportedColumnMapping(getColumnMappingMode(metadataEntry, protocolEntry));
+        cacheTableMetadataAsync(setCacheMetadata(metastoreTable.get(), Optional.ofNullable(metadataEntry.getDescription()), tableSnapshot.getVersion(), metadataEntry.getSchemaString()));
```

Comment:
I'm inclined to think that this operation should be performed in `io.trino.plugin.deltalake.DeltaLakeMetadata#getSnapshot(io.trino.spi.connector.ConnectorSession, io.trino.spi.connector.SchemaTableName, java.lang.String, java.util.Optional<java.lang.Long>)` when `atVersion` is empty in order to avoid caching multiple times the same table.